### ISSUE TITLE
Sensor Entity Names should be prefixed by device name

### ISSUE THREAD

**Issue by splitbrain** [garbled1/homeassistant_ecowitt#75]
When adding the integration, the device name can be configured. However the sensor names are simply added as ``sensor.daily_rain_rate`` it would be nice if they were automatically prefixed with the device name, eg. ``sensor.ecowitt_daily_rain_rate``.

Alternatively an optional entity prefix with an empty default might be a good way to add this in a backward compatible way.

(currently using the fork by @PoppyPop until this repo has been fixed)

**Comment by PoppyPop** [comment]
Hi, i checked the home assistant docs for the best practice, do you have some example ?
I checked, my production environnement and i have only 1 prefixed integration (OpenWeatherMap)

I someone knows an official guideline on sensor name, feel free to comment with a link

**Comment by tigers75** [comment]
The Fronius integration also does that, and it's very useful with a filter card *fronius* I can have all of the sensors displayed. When the integration changed names and added sensors, they were still displayed. 
Also it could be quite common to have a sensor.humidity or generic names like that. 
Don't know about official guidelines or so, but the change would be much appreciated. Of course it's something that has to be pondered, because it would break sensors historyhj

**Comment by mr-sneezy** [comment]
I agree with tigers75, it would help a lot to anyone setting up cards etc. I know it made it harder for me not having it prefixed. I'm not fussed about breaking the history data as I only store a month.

**Comment by adynis** [comment]
Yesss !! Exactly !! 
I have many temperature sensors ...  and one ugly example from ecowitt is: sensor.temperature_1 :( 
I would REALLY appreciate something like sensor.ecowitt_* as entity name (or, if there is any issue/concern, at least to be able to seelct this option during configuration :-) )

**Comment by DoerakkerHA** [comment]
hi all, are you aware a new version of the core is release including the ECOWITT integration?

now it does not provide the prefix but all sensors appear to be working fine with this new release https://www.home-assistant.io/integrations/ecowitt

maybe we should reconvene in the main support fora?

**Comment by tigers75** [comment]
I tired to install the official integration but didn't work for me. Also on the github I see there're some problems to resolve so, I'm waiting for the moment.

**Comment by sam-ward** [comment]
There was a problem with the .0 release which meant it didn’t work at all (at least for me), but that was resolved in .1.  There seems to be a smaller bug fix in .2 that was just released, but mines been working fine since .1.
